server_http.hpp

```diff
@@ -80,6 +80,7 @@ namespace SimpleWeb {
             std::string remote_endpoint_address;
             unsigned short remote_endpoint_port;
             
+            std::function<void(const std::exception&)> exception_handler;
```

Comment:
Maybe remove this, and move the try catch in Request::read_remote_endpoint_data to ServerBase:: read_request_and_content?